### PR TITLE
[#12] Linter and code formatter setup

### LINKED ISSUE #12: Linter and code formatter setup
**Is your feature request related to a problem? Please describe.**
We do not have any tools to enforce code style and formatting.

**Describe the solution you'd like**
We should integrate ESLint and Prettier into our workflow.

**Describe alternatives you've considered**
Other linters etc. exist but some members of the group have experience with ESLint and Prettier, so they are the best solution for us.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by seanhogunkim**
---
Set up base ESLint using AirBnB style guide
---

# Description:
Installed ESLint and AirBnb style guide to the project

### All Submissions:

* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you lint your code locally prior to submission?
3. [ ] Have you commented your code?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

## Instructions for setting up linting and formatting
Please set this up before making any contributions

1. Install ESLint and Prettier plugins for VS Code
    - https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
    - https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

2. `npm install` from root (from \Updog)
3. Navigate to `settings.json`
This can be done by doing Ctrl+Shift+P and typing 'open settings'
image
Open settings.json and add these lines somewhere in the file

```
"editor.defaultFormatter": "esbenp.prettier-vscode",
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
  }
```
Your settings.json may look something like this in the end

image
If it is all working correctly, ESLint analysis should pop up in JS files, and Prettier will format files properly upon saving. 

Reach out to devops if anything isn't working

Useful links:
https://enlear.academy/how-to-set-up-airbnb-style-guide-82413ea6c5f2
https://airbnb.io/javascript/
https://github.com/airbnb/javascript
https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
https://prettier.io/docs/en/configuration.html
https://github.com/prettier/eslint-config-prettier

### CODE REVIEW

**Comment by harristomy** [comment]
Could you add jest support to avoid errors being thrown for testing
image

File: package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "eslint-plugin-react": "^7.29.3",
+    "eslint-plugin-react-hooks": "^4.3.0",
+    "prettier": "2.5.1"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/seanhogunkim/Updog.git"
```
Comment (harristomy) [inline package.json]: refactor to main repo links? similar question on lines 26 and 28

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
```
Comment (harristomy) [inline .eslintrc.json]: Prevents ESLint throwing errors on testing files
`"jest": true `

**Review by harristomy — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 2 above)
Comment (seanhogunkim) [reply .eslintrc.json]: Added line

File: package.json (hunk 1 above)
Comment (seanhogunkim) [reply package.json]: Tried removing first

**Review by harristomy — APPROVED** [review]
Initial ESLint + Prettier setup looks good

**Review by harristomy — APPROVED** [review]

**Pull request merged**